mlflow/types/schema.py

```diff
@@ -524,6 +538,31 @@ def _merge(self, arr: "Array") -> "Array":
         raise MlflowException(f"Array type {self!r} and {arr!r} are incompatible")
 
 
+class SparkMLVector(Array):
+    """
+    Specification used to represent a vector type in Spark ML.
+    """
+
+    def __init__(self):
+        super().__init__(dtype=DataType.double)
+
+    def to_dict(self):
+        return {"type": SPARKML_VECTOR_TYPE}
+
+    @classmethod
+    def from_json_dict(cls, **kwargs):
+        return SparkMLVector()
+
+    def __repr__(self) -> str:
+        return "SparkML vector"
+
+    def __eq__(self, other) -> bool:
+        return isinstance(other, SparkMLVector)
+
+    def _merge(self, arr: "Array") -> "Array":
```

Comment:
```suggestion
    def _merge(self, arr: Array) -> Array:
```
nit: I guess it's fine to remove quotes here?

Btw is merging 2 vector types ok? If not, maybe we can adjust the message to `with another Array or SparkML vector type`).

Reply:
good point, merge another vector type is legitimate.